internal/security/secretstore/init.go

```diff
@@ -386,36 +388,114 @@ func (b *Bootstrap) BootstrapHandler(ctx context.Context, _ *sync.WaitGroup, _ s
 	}
 
 	// Add any additional services that need the known DB secret
+	lc.Infof("adding any additional services using redisdb for knownSecrets...")
 	services, ok := knownSecretsToAdd[redisSecretName]
 	if ok {
 		for _, service := range services {
-			configuration.Databases[service] = config.Database{
-				Service: service,
+			err = addServiceCredential(lc, redisSecretName, secretStore, service, redisCredentials)
+			if err != nil {
+				lc.Error(err.Error())
+				return false
 			}
 		}
 	}
 
+	lc.Infof("adding redisdb secret path for internal services...")
 	for _, info := range configuration.Databases {
 		service := info.Service
 
 		// add credentials to service path if specified and they're not already there
 		if len(service) != 0 {
-			err = addServiceCredential(lc, "redisdb", cred, service, redisCredentials)
+			err = addServiceCredential(lc, redisSecretName, secretStore, service, redisCredentials)
 			if err != nil {
 				lc.Error(err.Error())
 				return false
 			}
 		}
 	}
-
 	// security-bootstrapper-redis uses the path /v1/secret/edgex/security-bootstrapper-redis/ and go-mod-bootstrap
 	// with append the DB type (redisdb)
-	err = addDBCredential(lc, "security-bootstrapper-redis", cred, "redisdb", redisCredentials)
+	err = storeCredential(lc, "security-bootstrapper-redis", secretStore, redisSecretName, redisCredentials)
 	if err != nil {
 		lc.Error(err.Error())
 		return false
 	}
 
+	// for mqtt-bus creds
+	var mqttBusCredentials UserPasswordPair
+	if configuration.SecureMessageBus.Type == mqttSecretName {
+		mqttBusCredentials, err = getCredential("security-bootstrapper-mqtt", secretStore, mqttSecretName)
+		if err != nil {
+			if err != errNotFound {
+				lc.Error("failed to determine if MQTT credentials already exist or not: %w", err)
+				return false
+			}
+
+			lc.Info("Generating new password for MQTT bus")
+			mqttPassword, err := secretStore.GeneratePassword(ctx)
+			if err != nil {
+				lc.Error("failed to generate password for mqtt-bus")
+				return false
+			}
+
+			mqttBusCredentials = UserPasswordPair{
+				User:     defaultMqttUser,
+				Password: mqttPassword,
+			}
+		} else {
+			lc.Info("MQTT credentials already exist, skipping generating new password")
+		}
+
+		lc.Infof("adding any additional services using mqtt-bus for knownSecrets...")
+		services, ok := knownSecretsToAdd[mqttSecretName]
+		if ok {
+			for _, service := range services {
+				err = addServiceCredential(lc, mqttSecretName, secretStore, service, mqttBusCredentials)
+				if err != nil {
+					lc.Error(err.Error())
+					return false
+				}
+			}
+		}
+		err = storeCredential(lc, "security-bootstrapper-mqtt", secretStore, mqttSecretName, mqttBusCredentials)
+		if err != nil {
+			lc.Error(err.Error())
+			return false
+		}
+	}
```

Comment:
Can we make this more generic to Secure MessagBus rather than secure MQTT? i.e. we'll want to secure NATS in the near future.

Reply:
what is nats? without knowing the characteristics of nats, it is hard to make it generic... and i feel it is beyond the scope of this PR...

Reply:
NATS is a new implementation of our abstract MessageBus. 

The point is at this level in the code we need to be treating it as a Secure MessageBus without knowing the implementation. The messagebus_factory is where we put in the details needed for the specific implementation, currently Redis and Mosquitto

Reply:
yeah for bootstrapper, but this is in secretstore setup service ...kind of different route here

Reply:
That is my point. secretstore setup need to treat this as generic Secure MessageBus.